src/components/frame/Extensions/pages/ExtensionDetailsPage/utils.tsx

```diff
@@ -134,50 +149,85 @@ export const waitForDbAfterExtensionAction = ({
 
 export const getFormSuccessFn =
   <T extends FieldValues>({
-    setWaitingForEnableConfirmation,
+    setWaitingForActionConfirmation,
     extensionData,
     refetchColony,
     refetchExtensionData,
     navigate,
     colonyName,
   }: {
-    setWaitingForEnableConfirmation: SetStateFn;
+    setWaitingForActionConfirmation: SetStateFn;
     extensionData: AnyExtensionData;
     refetchColony: RefetchColonyFn;
     refetchExtensionData: RefetchExtensionDataFn;
     navigate: ReturnType<typeof useNavigate>;
     colonyName: string;
   }): OnSuccess<T> =>
-  async (_, { reset }) => {
-    setWaitingForEnableConfirmation(true);
+  async (fieldValues, { reset }) => {
+    setWaitingForActionConfirmation(true);
+    const isSaveChanges =
+      extensionData &&
+      isInstalledExtensionData(extensionData) &&
+      extensionData.extensionId === Extension.StakedExpenditure &&
+      extensionData.isEnabled;
+
     try {
       /* Wait for permissions first, so that the permissions warning doesn't flash in the ui */
       await waitForColonyPermissions({ extensionData, refetchColony });
-      await waitForDbAfterExtensionAction({
-        method: ExtensionMethods.ENABLE,
-        refetchExtensionData,
-      });
+      if (isSaveChanges) {
+        await waitForDbAfterExtensionAction({
+          method: ExtensionMethods.SAVE_CHANGES,
+          refetchExtensionData,
+          latestStakeFraction: convertFractionToWei(
+            fieldValues.params?.stakeFraction,
+          ),
+        });
+      } else {
+        await waitForDbAfterExtensionAction({
+          method: ExtensionMethods.ENABLE,
+          refetchExtensionData,
+        });
```

Comment:
Would be nice to make this more reusable for other extensions too.

Reply:
I'm not sure it's the right time to make this generic, no other extension offers updating parameters after enabling.